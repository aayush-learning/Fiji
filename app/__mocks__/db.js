const usersDB = {
  sync: () => {
    return {rows: []}
  },
  allDocs: () => {
    return {rows: []}
  }
}

export { usersDB }