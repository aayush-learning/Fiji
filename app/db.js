import PouchDB from 'pouchdb-react-native';

// these settings let us use cuids
// see: https://github.com/ericelliott/cuid/issues/54
global.navigator.mimeTypes = '';
global.navigator.userAgent = 'reactnative';    

const fijiDB = new PouchDB('fiji', { adapter: 'asyncstorage' });
console.log(fijiDB.adapter)
fijiDB.info().then(function (info) {
  console.log(info);
})
export default fijiDB