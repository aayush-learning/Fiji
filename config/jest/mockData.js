import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})