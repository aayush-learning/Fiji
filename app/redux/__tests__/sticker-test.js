jest.mock('../../db')

import { contentDB } from '../../db'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import sticker, {
  initialState,
  fetchStickersFailure,
  fetchStickersRequest,
  fetchStickersSuccess,
  fetchStickers,
  fetchStickerFailure,
  fetchStickerRequest,
  fetchStickerSuccess,
  fetchSticker,
  fetchStickerPacksFailure,
  fetchStickerPacksRequest,
  fetchStickerPacksSuccess,
  fetchStickerPacks,
} from '../sticker'
import { stickersData, stickerPacksData } from '../../../config/jest/mockData'

// test reducers

it('returns the same state on an unhandled action', () => {
  expect(sticker(initialState, { type: '_NULL' })).toMatchSnapshot()
})

it('handles FETCH_STICKERS_REQUEST action', () => {
  expect(sticker(initialState, fetchStickersRequest())).toMatchSnapshot()
})

it('handles FETCH_STICKERS_SUCCESS action', () => {
  expect(sticker(initialState, fetchStickersSuccess(stickersData))).toMatchSnapshot()
})

it('handles FETCH_STICKERS_FAILURE action', () => {
  expect(sticker(initialState, fetchStickersFailure())).toMatchSnapshot()
})

it('handles FETCH_STICKER_REQUEST action', () => {
  expect(sticker(initialState, fetchStickerRequest())).toMatchSnapshot()
})

it('handles FETCH_STICKER_SUCCESS action', () => {
  expect(sticker(initialState, fetchStickerSuccess(stickersData[0]))).toMatchSnapshot()
})

it('handles FETCH_STICKER_FAILURE action', () => {
  expect(sticker(initialState, fetchStickerFailure())).toMatchSnapshot()
})

it('handles FETCH_STICKER_PACKS_REQUEST action', () => {
  expect(sticker(initialState, fetchStickerPacksRequest())).toMatchSnapshot()
})

it('handles FETCH_STICKER_PACKS_SUCCESS action', () => {
  expect(sticker(initialState, fetchStickerPacksSuccess(stickersData))).toMatchSnapshot()
})

it('handles FETCH_STICKER_PACKS_FAILURE action', () => {
  expect(sticker(initialState, fetchStickerPacksFailure())).toMatchSnapshot()
})

// test actions

it('creates a FETCH_STICKERS_REQUEST action', () => {
  expect(fetchStickersRequest()).toMatchSnapshot()
})

it('creates a FETCH_STICKERS_SUCCESS action', () => {
  expect(fetchStickersSuccess(stickersData)).toMatchSnapshot()
})

it('creates a FETCH_STICKERS_FAILURE action', () => {
  expect(fetchStickersFailure()).toMatchSnapshot()
})

it('creates a FETCH_STICKER_REQUEST action', () => {
  expect(fetchStickerRequest()).toMatchSnapshot()
})

it('creates a FETCH_STICKER_SUCCESS action', () => {
  expect(fetchStickerSuccess(stickersData[0])).toMatchSnapshot()
})

it('creates a FETCH_STICKER_FAILURE action', () => {
  expect(fetchStickerFailure()).toMatchSnapshot()
})

it('creates a FETCH_STICKER_PACKS_REQUEST action', () => {
  expect(fetchStickerPacksRequest()).toMatchSnapshot()
})

it('creates a FETCH_STICKER_PACKS_SUCCESS action', () => {
  expect(fetchStickerPacksSuccess(stickerPacksData)).toMatchSnapshot()
})

it('creates a FETCH_STICKER_PACKS_FAILURE action', () => {
  expect(fetchStickerPacksFailure()).toMatchSnapshot()
})

// test thunks

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

it('handles fetchStickers action', async () => {
  const store = mockStore(initialState)
  contentDB.allDocs = jest.fn()
    .mockReturnValue({
      rows: stickersData.map(function (doc) {
        return { doc }
      })
    })
  await store.dispatch(fetchStickers('sticker-pack:caterpillar'))
  expect(store.getActions()).toMatchSnapshot()
})  

it('handles fetchSticker action', async () => {
  const store = mockStore(initialState)
  contentDB.get = jest.fn()
    .mockReturnValue(stickersData[0])
  await store.dispatch(fetchSticker(stickerPacksData[0]._id))
  expect(store.getActions()).toMatchSnapshot()
})  

it('handles fetchStickerPacks action', async () => {
  const store = mockStore(initialState)
  contentDB.allDocs = jest.fn()
    .mockReturnValue({
      rows: stickerPacksData.map(function (doc) {
        return { doc }
      })
    })
  await store.dispatch(fetchStickerPacks())
  expect(store.getActions()).toMatchSnapshot()
})  
