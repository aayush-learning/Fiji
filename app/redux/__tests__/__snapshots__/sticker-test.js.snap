// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a FETCH_STICKER_FAILURE action 1`] = `
Object {
  "type": "Fiji/game/FETCH_STICKER_FAILURE",
}
`;

exports[`creates a FETCH_STICKER_PACKS_FAILURE action 1`] = `
Object {
  "type": "Fiji/game/FETCH_STICKER_PACKS_FAILURE",
}
`;

exports[`creates a FETCH_STICKER_PACKS_REQUEST action 1`] = `
Object {
  "type": "Fiji/game/FETCH_STICKER_PACKS_REQUEST",
}
`;

exports[`creates a FETCH_STICKER_PACKS_SUCCESS action 1`] = `
Object {
  "packs": Array [],
  "type": "Fiji/game/FETCH_STICKER_PACKS_SUCCESS",
}
`;

exports[`creates a FETCH_STICKER_REQUEST action 1`] = `
Object {
  "type": "Fiji/game/FETCH_STICKER_REQUEST",
}
`;

exports[`creates a FETCH_STICKER_SUCCESS action 1`] = `
Object {
  "sticker": undefined,
  "type": "Fiji/game/FETCH_STICKER_SUCCESS",
}
`;

exports[`creates a FETCH_STICKERS_FAILURE action 1`] = `
Object {
  "type": "Fiji/game/FETCH_STICKERS_FAILURE",
}
`;

exports[`creates a FETCH_STICKERS_REQUEST action 1`] = `
Object {
  "type": "Fiji/game/FETCH_STICKERS_REQUEST",
}
`;

exports[`creates a FETCH_STICKERS_SUCCESS action 1`] = `
Object {
  "stickers": Array [],
  "type": "Fiji/game/FETCH_STICKERS_SUCCESS",
}
`;

exports[`handles FETCH_STICKER_FAILURE action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKER_PACKS_FAILURE action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKER_PACKS_REQUEST action 1`] = `
Object {
  "isFetching": true,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKER_PACKS_SUCCESS action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKER_REQUEST action 1`] = `
Object {
  "isFetching": true,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKER_SUCCESS action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": undefined,
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKERS_FAILURE action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKERS_REQUEST action 1`] = `
Object {
  "isFetching": true,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles FETCH_STICKERS_SUCCESS action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;

exports[`handles fetchSticker action 1`] = `
Array [
  Object {
    "type": "Fiji/game/FETCH_STICKER_REQUEST",
  },
  Object {
    "sticker": Object {
      "_id": "sticker:caterpillar:walk",
      "svg": "PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pg0KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDE5LjAuMCwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPg0KPHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJMYXllcl8xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB4PSIwcHgiIHk9IjBweCINCgkgdmlld0JveD0iMCAwIDUxMiA1MTIiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDUxMiA1MTI7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4NCjxwYXRoIHN0eWxlPSJmaWxsOiM0ODk4REE7IiBkPSJNMzY0LjA0MiwzMjkuMTg1YzEzLjIwNS04LjEwMywyOS4xNjUtMTIuNzYxLDQ2LjA2OC0xMi43NjFjNDkuNzIzLDAsODkuOTYxLDQwLjMxMiw4OS45NjEsOTAuMDM1DQoJcy00MC4xOTksOTAuMDM1LTg5LjkyMiw5MC4wMzVjLTE2LjkwMywwLTMyLjktNC42NTgtNDYuMTA1LTEyLjc2MVYzMjkuMTg1SDM2NC4wNDJ6Ii8+DQo8cGF0aCBzdHlsZT0iZmlsbDojRjhBODA1OyIgZD0iTTI3Mi44MDcsMzI4LjQ3OGMxMy4yMDUtNy42NTksMjguNDY5LTEyLjA1Myw0NC44NjgtMTIuMDUzYzE2LjkwMywwLDMyLjYzOCw0LjY1OCw0Ni4xNDQsMTIuNzYxDQoJYzI2LjI1NCwxNS43MjYsNDMuNzgsNDQuNDUzLDQzLjc4LDc3LjI3NHMtMTcuNTgyLDYxLjU0OC00My44MzYsNzcuMjc0Yy0xMy41MDUsOC4xMDMtMjkuMTc1LDEyLjc2MS00Ni4wNzgsMTIuNzYxDQoJYy0xNi4zOTgsMC0zMS42NzItNC4zOTQtNDQuODc3LTEyLjA1M1YzMjguNDc4eiIvPg0KPHBhdGggc3R5bGU9ImZpbGw6IzNFQUY2NDsiIGQ9Ik0yMjMuODUxLDMxNi41MDljMS4yMTItMC4wNiwyLjQyNS0wLjA4NCwzLjYzNy0wLjA4NGMxNi4zOTgsMCwzMS43NzYsNC4zOTQsNDUuMDE4LDEyLjA1Mw0KCWMyNi45MTUsMTUuNTU4LDQ1LjAxOCw0NC42NTcsNDUuMDE4LDc3Ljk4MnMtMTguMTAzLDYyLjQyNC00NS4wMTgsNzcuOTgyYy0xMy4yNDEsNy42NTktMjguNjE5LDEyLjA1My00NS4wMTgsMTIuMDUzDQoJYy0xNy4xNDMsMC0zMy4xODEtNC44MDItNDYuODE4LTEzLjEyMUwyMjMuODUxLDMxNi41MDl6Ii8+DQo8cGF0aCBzdHlsZT0iZmlsbDojRDI1MjU2OyIgZD0iTTIxMi41MDcsMzYyLjg0MmM3LjI1MSwxMy4yMDUsMTEuMzgsMjcuODExLDExLjM4LDQzLjcxN2MwLDMyLjU4MS0xNy4yOTksNjEuMDY3LTQzLjIxNyw3Ni44NjQNCgljLTEzLjYzNyw4LjMxOS0yOS42NzYsMTMuMDk2LTQ2LjgxOCwxMy4wOTZjLTQ5LjcyMywwLTkwLjAzNS00MC4yMjQtOTAuMDM1LTg5Ljk0OGMwLTE1LjkwNiw0LjEzLTMwLjUyNCwxMS4zOC00My43MjlIMjEyLjUwN3oiLz4NCjxwYXRoIHN0eWxlPSJmaWxsOiM0ODk4REE7IiBkPSJNMjEyLjE3MSwyNzQuMDA4YzcuMDk1LDEzLjIwNSwxMS4yOTYsMjcuMDA3LDExLjY4MSw0Mi4zMDFjMC4wMjQsMC43NjgsMC4wMzYsMS42MzYsMC4wMzYsMi40MTcNCgljMCwxNS45MDYtNC4xMywzMC45MDItMTEuMzgsNDMuODY4Yy0xNC44MjYsMjYuNTktNDIuNzM3LDQ0Ljg5OS03NS4wMjksNDYuMTcxYy0xLjIsMC4wNDgtMi40MTMsMC4wODQtMy42MjUsMC4wODQNCgljLTMzLjgxNywwLTYzLjI3Ny0xOC42MzctNzguNjU1LTQ2LjIxMmMtNy4yNTEtMTIuOTY1LTExLjM4LTI4LjEwOC0xMS4zOC00NC4wMTRjMC0xNi4xNTgsNC4yNjItMzEuNDA5LDExLjcxNy00NC42MTRIMjEyLjE3MXoiLz4NCjxwYXRoIHN0eWxlPSJmaWxsOiNGOEE4MDU7IiBkPSJNMjIzLjg4NywyMjkuOTljMCwxNi4xNTgtNC4yNjIsMzEuMzA4LTExLjcxNyw0NC40MTdjLTE1LjQ3NCwyNy4yMzktNDQuNzUzLDQ1LjYxOC03OC4zMTksNDUuNjE4DQoJcy02Mi44NDUtMTguMzc5LTc4LjMxOS00NS42MThjLTcuNDU1LTEzLjEwOS0xMS43MTctMjguMjU5LTExLjcxNy00NC40MTdjMC00OS43MjMsNDAuMzEyLTkwLjAzNSw5MC4wMzUtOTAuMDM1DQoJUzIyMy44ODcsMTgwLjI2OCwyMjMuODg3LDIyOS45OXoiLz4NCjxnPg0KCTxwYXRoIHN0eWxlPSJmaWxsOiNFQ0YwRjE7IiBkPSJNNjAuMDIzLDExMS41NDVjLTI2LjQ3OCwwLTQ4LjAxOS0yMS41NDEtNDguMDE5LTQ4LjAxOXMyMS41NDEtNDguMDE5LDQ4LjAxOS00OC4wMTkNCgkJczQ4LjAxOSwyMS41NDEsNDguMDE5LDQ4LjAxOVM4Ni41MDEsMTExLjU0NSw2MC4wMjMsMTExLjU0NXoiLz4NCgk8cGF0aCBzdHlsZT0iZmlsbDojRUNGMEYxOyIgZD0iTTIwNy42ODEsMTExLjU0NWMtMjYuNDc4LDAtNDguMDE5LTIxLjU0MS00OC4wMTktNDguMDE5czIxLjU0MS00OC4wMTksNDguMDE5LTQ4LjAxOQ0KCQlTMjU1LjcsMzcuMDQ4LDI1NS43LDYzLjUyNlMyMzQuMTU5LDExMS41NDUsMjA3LjY4MSwxMTEuNTQ1eiIvPg0KPC9nPg0KPGc+DQoJPHBhdGggc3R5bGU9ImZpbGw6IzIzMUYyMDsiIGQ9Ik00MDkuOTYsMzA0LjQxOWMtMTYuMTI4LDAtMzEuOTY2LDMuODA3LTQ2LjIxOCwxMS4wNTZjLTE0LjI1Mi03LjI1LTMwLjA5LTExLjA1Ni00Ni4yMTgtMTEuMDU2DQoJCWMtMTUuNjQyLDAtMzEuMDcxLDMuNi00NS4wMTgsMTAuNDU3Yy0xMS43NDEtNS43NzItMjQuNTI5LTkuMjI4LTM3LjYxMy0xMC4xNzljLTEuNDY5LTEwLjQ5OC00LjU2Ny0yMC43NDUtOS4xNzUtMzAuMjgzDQoJCWM2LjY3MS0xMy43ODcsMTAuMTc0LTI5LjAxMiwxMC4xNzQtNDQuNDI1YzAtNDMuNjItMjcuNTIxLTgwLjkxOS02Ni4xMDQtOTUuNDg4bDE1LjM2LTE1LjM2DQoJCWM2Ljk2MywyLjgzMiwxNC41NjYsNC40MDYsMjIuNTM0LDQuNDA2YzMzLjA5NywwLDYwLjAyMy0yNi45MjcsNjAuMDIzLTYwLjAyM1MyNDAuNzc5LDMuNTAxLDIwNy42ODIsMy41MDENCgkJcy02MC4wMjMsMjYuOTI3LTYwLjAyMyw2MC4wMjNjMCwxNi4zMjgsNi41NjEsMzEuMTQ1LDE3LjE3NSw0MS45NzhsLTIyLjgwNSwyMi44MDVjLTIuNy0wLjIxNS01LjQyMS0wLjM1Ny04LjE3NS0wLjM1Nw0KCQljLTMuNzg3LDAtNy41MjMsMC4yMjQtMTEuMjA1LDAuNjI4bC0yMS40NDYtMjEuNDQ1YzExLjU5MS0xMC45NDksMTguODQ1LTI2LjQ0MywxOC44NDUtNDMuNjA4DQoJCWMwLTMzLjA5Ny0yNi45MjctNjAuMDIzLTYwLjAyMy02MC4wMjNTMCwzMC40MjksMCw2My41MjZzMjYuOTI3LDYwLjAyMyw2MC4wMjMsNjAuMDIzYzcuMDY2LDAsMTMuODQ2LTEuMjM2LDIwLjE0OS0zLjQ5DQoJCWwxNS4zNzksMTUuMzc5Yy0zNy4zMzcsMTUuMTc5LTYzLjczOSw1MS44My02My43MzksOTQuNTUxYzAsMTUuNDEyLDMuNTAyLDMwLjYzNSwxMC4xNyw0NC40MTcNCgkJYy02LjY2OSwxMy43ODUtMTAuMTcsMjkuMDA4LTEwLjE3LDQ0LjQxN2MwLDE1LjE5MiwzLjQwMywzMC4yMTIsOS44ODEsNDMuODE3Yy02LjQ3OSwxMy42MTItOS44ODEsMjguNjI5LTkuODgxLDQzLjgxNw0KCQljMCw1Ni4yNjUsNDUuNzc1LDEwMi4wNCwxMDIuMDQsMTAyLjA0YzE2LjM1OSwwLDMyLjQwNC0zLjkxMiw0Ni44MTgtMTEuMzY0YzE0LjQxNiw3LjQ1MSwzMC40NiwxMS4zNjQsNDYuODE4LDExLjM2NA0KCQljMTUuNjQzLDAsMzEuMDcyLTMuNiw0NS4wMTgtMTAuNDU3YzEzLjk0Nyw2Ljg1NywyOS4zNzUsMTAuNDU3LDQ1LjAxOCwxMC40NTdjMTYuMTMxLDAsMzEuOTcxLTMuODA4LDQ2LjIxOS0xMS4wNTYNCgkJYzE0LjI1Miw3LjI1LDMwLjA4OSwxMS4wNTYsNDYuMjE3LDExLjA1NmM1Ni4yNjUsMCwxMDIuMDQtNDUuNzc1LDEwMi4wNC0xMDIuMDRTNDY2LjIyNSwzMDQuNDE5LDQwOS45NiwzMDQuNDE5eiBNMjA3LjY4MSwyNy41MTINCgkJYzE5Ljg1OCwwLDM2LjAxNCwxNi4xNTYsMzYuMDE0LDM2LjAxNFMyMjcuNTM5LDk5LjU0LDIwNy42ODEsOTkuNTRzLTM2LjAxNC0xNi4xNTYtMzYuMDE0LTM2LjAxNFMxODcuODIzLDI3LjUxMiwyMDcuNjgxLDI3LjUxMnoNCgkJIE0yNC4wMDksNjMuNTI2YzAtMTkuODU4LDE2LjE1Ni0zNi4wMTQsMzYuMDE0LTM2LjAxNHMzNi4wMTQsMTYuMTU2LDM2LjAxNCwzNi4wMTRjMCwxMy41MDMtNy40NzQsMjUuMjg3LTE4LjQ5NywzMS40NTENCgkJYy0wLjI2MSwwLjEyOC0wLjUxNSwwLjI2OS0wLjc2NywwLjQxN2MtNS4wMDcsMi42NDItMTAuNzA1LDQuMTQ2LTE2Ljc1LDQuMTQ2QzQwLjE2NSw5OS41NCwyNC4wMDksODMuMzgzLDI0LjAwOSw2My41MjZ6DQoJCSBNNjUuNjc1LDM1Ni43ODJjLTYuNDQ3LTExLjUyNS05Ljg1My0yNC42NDktOS44NTMtMzcuOTU2YzAtNi44MTcsMC44OTgtMTMuNTg2LDIuNjM3LTIwLjExOQ0KCQljMTkuMTIzLDIwLjkzNyw0Ni40MzUsMzMuMzI0LDc1LjM5MywzMy4zMjRjMjguOTY2LDAsNTYuMjg1LTEyLjM5NCw3NS40MDktMzMuMzQxYzEuNTYyLDUuODcyLDIuNDM1LDExLjk0NywyLjU5MSwxOC4yMDMNCgkJYzAuMDIyLDAuNjQsMC4wMywxLjI4MSwwLjAzLDEuOTMyYzAsMTMuMzA3LTMuNDA3LDI2LjQzMy05Ljg2MiwzNy45NzFjLTEzLjM2MiwyMy45NjctMzcuNjY3LDM4LjkxOS02NS4wMzcsMzkuOTk3DQoJCWMtMS4wMzcsMC4wNDItMi4wODQsMC4wNjItMy4xMzIsMC4wNjJDMTA1LjU4NCwzOTYuODU1LDc5LjQ2MywzODEuNTA1LDY1LjY3NSwzNTYuNzgyeiBNMTMzLjg1MiwxNTEuOTYNCgkJYzQzLjAyNiwwLDc4LjAzLDM1LjAwNCw3OC4wMyw3OC4wM2MwLDEzLjUwMy0zLjUwOSwyNi44MS0xMC4xNDksMzguNDg4Yy0xMy44NTgsMjQuMzktMzkuODY5LDM5LjU0Mi02Ny44ODIsMzkuNTQyDQoJCXMtNTQuMDIzLTE1LjE1Mi02Ny44ODQtMzkuNTQ3Yy02LjYzNy0xMS42NzMtMTAuMTQ2LTI0Ljk4MS0xMC4xNDYtMzguNDgzQzU1LjgyMiwxODYuOTY1LDkwLjgyNiwxNTEuOTYsMTMzLjg1MiwxNTEuOTZ6DQoJCSBNNTUuODIyLDQwNi40NTljMC02LjQ4OSwwLjgxNi0xMi45MzEsMi4zOTQtMTkuMTY5YzE5LjExNywyMS4xMDEsNDYuNTAxLDMzLjU3NSw3NS42MzcsMzMuNTc1YzEuMzkzLDAsMi43NzctMC4wMjgsNC4wOTctMC4wODINCgkJYzI3LjQ2OC0xLjA4Miw1My4zMTktMTMuNDkzLDcxLjUzNC0zMy41MjFjMS41ODIsNi4yNDYsMi40LDEyLjY5OSwyLjQsMTkuMTk3YzAsMjcuNDQ0LTE0LjAwMyw1Mi4zNjYtMzcuNDY0LDY2LjY2Ng0KCQljLTEyLjE4Niw3LjQzNS0yNi4yMTUsMTEuMzY1LTQwLjU2NiwxMS4zNjVDOTAuODI2LDQ4NC40ODksNTUuODIyLDQ0OS40ODUsNTUuODIyLDQwNi40NTl6IE0yMDMuNzgzLDQ4MC44MTQNCgkJYzIwLjM4NC0xOS4wNiwzMi4xMDktNDUuNjQ3LDMyLjEwOS03NC4zNTVjMC0xNS4xODgtMy40MDItMzAuMjA1LTkuODgxLTQzLjgxN2M1LjA1LTEwLjYxMSw4LjIyOC0yMi4wNzUsOS4zODUtMzMuODA5DQoJCWMxMC45NDksMS4xMTMsMjEuNTcyLDQuNTI1LDMxLjEwMiwxMC4wMzZjMjQuMDY5LDEzLjkxMywzOS4wMjEsMzkuODEyLDM5LjAyMSw2Ny41OWMwLDI3Ljc3OC0xNC45NTIsNTMuNjc3LTM5LjAyNCw2Ny41OTENCgkJYy0xMS44MDgsNi44MjktMjUuMjk2LDEwLjQzOS0zOS4wMDcsMTAuNDM5QzIxOS4zOTMsNDg0LjQ4OSwyMTEuNCw0ODMuMjM5LDIwMy43ODMsNDgwLjgxNHogTTI5Ni40ODYsNDgxLjU5OA0KCQljMjAuNzU3LTE5LjExOSwzMy4wNDMtNDYuMzUzLDMzLjA0My03NS4xMzdjMC0yOC43ODYtMTIuMjg3LTU2LjAyLTMzLjA0My03NS4xMzdjNi44MTUtMS45MDQsMTMuODk1LTIuODkzLDIxLjAzOC0yLjg5Mw0KCQljMTQuMTQ2LDAsMjcuOTkzLDMuODIxLDQwLjA0OSwxMS4wNTVjMC4wMTYsMC4wMSwwLjAzMSwwLjAxOSwwLjA0NywwLjAyOWMwLjAwNiwwLjAwNCwwLjAxMiwwLjAwOCwwLjAxOSwwLjAxMg0KCQljMjMuNzQzLDE0LjI1MywzNy45MTYsMzkuMjY5LDM3LjkxNiw2Ni45MzVjMCwyNy42Ni0xNC4xNjcsNTIuNjcxLTM3LjkwNyw2Ni45M2MtMC4wMTYsMC4wMS0wLjAzMSwwLjAyLTAuMDQ4LDAuMDMNCgkJYy0wLjAxMSwwLjAwNi0wLjAyMiwwLjAxNC0wLjAzNCwwLjAyYy0xMi4wNDksNy4yMjktMjUuODk1LDExLjA1LTQwLjA0MiwxMS4wNUMzMTAuMzgxLDQ4NC40ODksMzAzLjMwMSw0ODMuNTAxLDI5Ni40ODYsNDgxLjU5OHoNCgkJIE00MDkuOTYsNDg0LjQ4OWMtNy44MDQsMC0xNS41MTYtMS4xNjktMjIuODg3LTMuNDI2YzIwLjQwMi0xOS4xMTMsMzIuNDkxLTQ2LjE4MywzMi40OTEtNzQuNjA0DQoJCWMwLTI4LjQyMy0xMi4wOS01NS40OTQtMzIuNDg5LTc0LjYwNGM3LjM3MS0yLjI1NywxNS4wODEtMy40MjYsMjIuODg2LTMuNDI2YzQzLjAyNiwwLDc4LjAzLDM1LjAwNCw3OC4wMyw3OC4wMw0KCQlTNDUyLjk4Niw0ODQuNDg5LDQwOS45Niw0ODQuNDg5eiIvPg0KCTxjaXJjbGUgc3R5bGU9ImZpbGw6IzIzMUYyMDsiIGN4PSI2MC4wMjMiIGN5PSI2My41MjYiIHI9IjEyLjAwNSIvPg0KCTxjaXJjbGUgc3R5bGU9ImZpbGw6IzIzMUYyMDsiIGN4PSIyMDcuNjgxIiBjeT0iNjMuNTI2IiByPSIxMi4wMDUiLz4NCgk8cGF0aCBzdHlsZT0iZmlsbDojMjMxRjIwOyIgZD0iTTEzMy44NTIsMjg0LjQxMmMyNC44MjQsMCw0Ni45NzgtMTcuNyw1Mi42NzgtNDIuMDg2YzEuNTA5LTYuNDU2LTIuNTAxLTEyLjkxMy04Ljk1Ny0xNC40MjINCgkJYy02LjQ2My0xLjUxNi0xMi45MTMsMi41MDItMTQuNDIyLDguOTU3Yy0zLjEzNywxMy40MjEtMTUuNzMyLDIzLjU0MS0yOS4yOTksMjMuNTQxcy0yNi4xNjItMTAuMTIxLTI5LjI5OS0yMy41NDENCgkJYy0xLjUxLTYuNDU3LTcuOTY4LTEwLjQ2NC0xNC40MjItOC45NTdjLTYuNDU2LDEuNTA5LTEwLjQ2Niw3Ljk2Ni04Ljk1NywxNC40MjJDODYuODc0LDI2Ni43MTIsMTA5LjAyOCwyODQuNDEyLDEzMy44NTIsMjg0LjQxMnoiDQoJCS8+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8L3N2Zz4NCg==",
    },
    "type": "Fiji/game/FETCH_STICKER_SUCCESS",
  },
]
`;

exports[`handles fetchStickerPacks action 1`] = `
Array [
  Object {
    "type": "Fiji/game/FETCH_STICKER_PACKS_REQUEST",
  },
  Object {
    "packs": Array [
      "caterpillar.svg",
      "cheshire-cat.svg",
    ],
    "type": "Fiji/game/FETCH_STICKER_PACKS_SUCCESS",
  },
]
`;

exports[`handles fetchStickers action 1`] = `
Array [
  Object {
    "type": "Fiji/game/FETCH_STICKERS_REQUEST",
  },
  Object {
    "stickers": undefined,
    "type": "Fiji/game/FETCH_STICKERS_SUCCESS",
  },
]
`;

exports[`returns the same state on an unhandled action 1`] = `
Object {
  "isFetching": false,
  "packs": Array [],
  "sticker": Object {},
  "stickers": Array [],
}
`;
